test/integration/usages/try_variables.rb

```diff
@@ -0,0 +1,23 @@
+module Liquid
+  class Context
+    remove_method :try_variable_find_in_environments
+    def try_variable_find_in_environments(key, raise_on_not_found:)
+      Usage.track("Using try_variable_find_in_environment")
+      @environments.each do |environment|
+        found_variable = lookup_and_evaluate(environment, key, raise_on_not_found: raise_on_not_found)
+        if !found_variable.nil? || @strict_variables && raise_on_not_found
+          return found_variable
+        end
+        Usage.track("try_variable_find_in_environment reports Nil but responds to key") if environment.key?(key)
```

Comment:
Are we worried about `Object#nil?` or `Nil#key?` being overridden?

Reply:
This wasn't really a real test even though it will be. The tracking will be similar to this and its to confirm if nil values are getting rejected or if found_variable is always non-nil if a key is present.

Reply:
Oh right, this is just in the integration tests.

In that case, we should make it even less coupled to the implementation itself and be able to unpatch the method.  For both those reasons, we should use `alias_method` to preserve the original method, then we can wrap the method with this patch and use the original method to unpatch the method. We don't want other integration tests to be actually testing this implementation instead of the real implementation of `try_variable_find_in_environments`.

Reply:
Yeah agree with this, I've just been working a little backwards on this. `alias_method` makes so much more sense here as it provides an easy way to enable and disable. I'll play around with this approach and see how it can work